tests/e2e/test_pvc_creation_scale.py

```diff
@@ -0,0 +1,164 @@
+
+
+@scale
+@ignore_leftovers
+class TestPVCCreationScale(E2ETest):
+    """
+    Base class for PVC scale creation
+    """
+    @pytest.fixture()
+    def namespace(self, project_factory):
+        """
+        Create a new project
+        """
+        proj_obj = project_factory()
+        self.namespace = proj_obj.namespace
+
+    @pytest.mark.parametrize(
+        argnames=["access_mode", "interface"],
+        argvalues=[
+            pytest.param(
+                *[constants.ACCESS_MODE_RWO, constants.CEPHBLOCKPOOL],
+                marks=pytest.mark.polarion_id("OCS-1225")
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWX, constants.CEPHBLOCKPOOL],
+                marks=pytest.mark.polarion_id("OCS-2010")
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWO, constants.CEPHFS_INTERFACE],
+                marks=pytest.mark.polarion_id("OCS-2011")
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWX, constants.CEPHFS_INTERFACE],
+                marks=pytest.mark.polarion_id("OCS-2008")
+            ),
+        ]
+    )
+    @pytest.mark.usefixtures(namespace.__name__)
+    def test_1500_pvc_creation_scale(
+        self, namespace, teardown_factory, access_mode, interface
+    ):
+        """
+        Measuring PVC creation time while scaling PVC
+        """
+        number_of_pvc = 1500
+        log.info(f"Start creating {access_mode}-{interface} {number_of_pvc} PVC")
+
+        if interface == constants.CEPHBLOCKPOOL:
+            self.sc_obj = constants.DEFAULT_STORAGECLASS_RBD
+        elif interface == constants.CEPHFS_INTERFACE:
+            self.sc_obj = constants.DEFAULT_STORAGECLASS_CEPHFS
+
+        pvc_objs = helpers.create_multiple_pvcs(
+            sc_name=self.sc_obj,
+            namespace=self.namespace,
+            number_of_pvc=number_of_pvc,
+            size=f"{random.randrange(5, 105, 5)}Gi",
+            access_mode=access_mode
+        )
+        for pvc_obj in pvc_objs:
+            pvc_obj.reload()
+            teardown_factory(pvc_obj)
+        pvc_name = list()
+        with ThreadPoolExecutor(max_workers=5) as executor:
+            for pvc_obj in pvc_objs:
+                executor.submit(
+                    helpers.wait_for_resource_state, pvc_obj,
+                    constants.STATUS_BOUND
+                )
+
+                executor.submit(pvc_obj.reload)
+                pvc_name.append(pvc_obj.name)
+
+        pvc_create_time = helpers.measure_pvc_creation_time_bulk(
+            interface=interface, pvc_name_list=pvc_name
+        )
+
+        # TODO: Update below code with google API, to record value in spreadsheet
+        # TODO: For now observing Google API limit to write more than 100 writes
+        csv_obj = csv.writer(open(f"/tmp/{self.sc_obj}-{access_mode}.csv", "w"))
+        for k, v in pvc_create_time.items():
+            csv_obj.writerow([k, v])
+
```

Comment:
This test case require all 4 PVC type, due to this there are some similar code lines but with different interfaces.